src/Microsoft.Crank.Agent/Controllers/JobsController.cs

```diff
@@ -332,6 +332,64 @@ public async Task<IActionResult> UploadAttachment(int id)
             return Ok();
         }
 
+        [HttpPost("{id}/attachment/zip")]
+        [RequestSizeLimit(10_000_000_000)]
+        [RequestFormLimits(MultipartBodyLengthLimit = 10_000_000_000)]
+        public async Task<IActionResult> UploadAttachmentZip(int id)
+        {
+            // Because uploaded files can be big, the client is supposed to keep
+            // pinging the service to keep the job alive during the Initialize state.
+
+            var destinationFilename = Request.Headers["destinationFilename"].ToString();
+
+            Log.Info($"Uploading archive: {destinationFilename}");
+
+            var job = _jobs.Find(id);
+
+            if (job == null)
+            {
+                return NotFound("Job doesn't exist anymore");
+            }
+
+            if (job.State != JobState.Initializing)
+            {
+                Log.Info($"Attachment rejected, job is {job.State}");
+                return StatusCode(500, $"The job can't accept attachment as its state is {job.State}");
```

Comment:
I agree that it's worth logging the current state of the job, but returning it to the caller may not be a good idea. Maybe simply respond that it's not in the `initializing` state and do not disclose the current state?

Reply:
It could be important for debugging to understand at what point the client tried to upload a file and how we got into this state. There is nothing to hide here.